### ISSUE TITLE
Call to riak_repl_wm_stats:jsonify_stats fails during fullsync

### ISSUE THREAD

**Issue by fprimex** [basho/riak_repl#173]
See ZD ticket 3634 for client detail. When a fullsync is running, calls to riak_repl_wm_stats:jsonify_stats will fail with e.g.

{error,{error,function_clause,[{riak_repl_wm_stats,jsonify_stats,[[{fullsync,63,left},...

This is called during webmachine calls to '/riak-repl/stats'.

**Comment by fprimex** [comment]
This ticket is being tracked by clients at: https://github.com/basho/riak_ee-issues/issues/1

**Comment by bookshelfdave** [comment]
Hitting /stats during fullsync doesn't seem to be the issue, as I can see the stats fine w/ a 1.2.0 build. I'll dig further.

[1]R15B01:[(riak_ee-1.2.0)]prime:~/basho/riak_ee$ curl -q http://localhost:8091/riak-repl/stats | python -mjson.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1184  100  1184    0     0   258k      0 --:--:-- --:--:-- --:--:--  385k
{
    "client_bytes_recv": 0, 
    "client_bytes_sent": 0, 
    "client_connect_errors": 0, 
    "client_connects": 0, 
    "client_redirect": 0, 
    "client_rx_kbps": [], 
    "client_stats": [], 
    "client_tx_kbps": [], 
    "elections_elected": 1, 
    "elections_leader_changed": 1, 
    "leader": "dev1@127.0.0.1", 
    "leader_garbage_collection": {
        "fullsweep_after": 0, 
        "min_bin_vheap_size": 46368, 
        "min_heap_size": 233, 
        "minor_gcs": 0
    }, 
    "leader_heap_size": 233, 
    "leader_message_queue_len": 0, 
    "leader_reductions": 6532, 
    "leader_stack_size": 9, 
    "leader_total_heap_size": 233, 
    "local_leader_heap_size": 233, 
    "local_leader_message_queue_len": 0, 
    "objects_dropped_no_clients": 0, 
    "objects_dropped_no_leader": 0, 
    "objects_forwarded": 0, 
    "objects_sent": 0, 
    "server_bytes_recv": 103260, 
    "server_bytes_sent": 3732, 
    "server_connect_errors": 0, 
    "server_connects": 0, 
    "server_fullsyncs": 0, 
    "server_rx_kbps": [], 
    "server_stats": [
        {
            "message_queue_len": 0, 
            "pid": "<0.1790.0>", 
            "status": {
                "dropped_count": 0, 
                "fullsync": 159851741583067506678528028578343455274867621888, 
                "fullsync_worker": "<0.1791.0>", 
                "get_pool_size": 5, 
                "node": "dev1@127.0.0.1", 
                "partition_start": 0.06, 
                "queue_byte_size": 0, 
                "queue_length": 0, 
                "site": "foo", 
                "stage_start": 0.06, 
                "state": "build_keylist", 
                "strategy": "riak_repl_keylist_server"
            }
        }
    ], 
    "server_tx_kbps": []
}

**Comment by bookshelfdave** [comment]
Ok, I have a patch for this bug. I'm not sure how to deliver this, as the repl 1.2 branch has the **1.2.1** changes, and it looks like Cisco has **1.2.0**.

```
diff --git a/src/riak_repl_wm_stats.erl b/src/riak_repl_wm_stats.erl
index 01d1603..134cd60 100644
--- a/src/riak_repl_wm_stats.erl
+++ b/src/riak_repl_wm_stats.erl
@@ -99,6 +99,8 @@ format_stats(Type, [{P, M, {status, S}}|T], Acc) ->
                             M, {status, jsonify_stats(S, [])}]|Acc]).
 jsonify_stats([], Acc) ->
     lists:flatten(lists:reverse(Acc));
+jsonify_stats([{fullsync, Num, _Left}|T], Acc) ->
+    jsonify_stats(T, [{"partitions_left", Num} | Acc]);
 jsonify_stats([{K,V}|T], Acc) when is_pid(V) ->
     jsonify_stats(T, [{K,list_to_binary(erlang:pid_to_list(V))}|Acc]);
 jsonify_stats([{K,V}|T], Acc) when is_list(V) ->
```

(dev1@127.0.0.1)1> Foo = [{fullsync,63,left},{partition,251195593916248939066258330623111144003363405824},{partition_start,0.88},{stage_start,0.88}],[{state,request_partition},{connected,<<"10.252.233.7:19010">>},{put_pool_size,5},{fullsync_worker,<<"<15426.1297.0>">>},{strategy,riak_repl_keylist_client},{site,<<"sj-bts-cluster">>},{node,'riak2@10.240.208.198'}].
[{state,request_partition},
 {connected,<<"10.252.233.7:19010">>},
 {put_pool_size,5},
 {fullsync_worker,<<"<15426.1297.0>">>},
 {strategy,riak_repl_keylist_client},
 {site,<<"sj-bts-cluster">>},
 {node,'riak2@10.240.208.198'}]

(dev1@127.0.0.1)2> riak_repl_wm_stats:jsonify_stats(Foo, []).
[{"partitions_left",63},
 {partition,251195593916248939066258330623111144003363405824},
 {partition_start,0.88},
 {stage_start,0.88}]

**Comment by bookshelfdave** [comment]
Merged for 1.3. Any previous version will require a patch.
